### PR TITLE
Get rid of quickcheck

### PR DESCRIPTION
**Pull request by kurnevsky**

### CODE REVIEW

File: src/toxcore/state_format/old.rs [hunk 1]
```diff
@@ -43,7 +43,7 @@ pub const NOSPAMKEYSBYTES: usize = NOSPAMBYTES + PUBLICKEYBYTES + SECRETKEYBYTES
 /// The `Default` implementation generates random `NospamKeys`.
 impl Default for NospamKeys {
     fn default() -> Self {
-        let nospam = NoSpam::default();
+        let nospam = NoSpam::new();
```
Comment (kpp) [inline src/toxcore/state_format/old.rs]: Actually `new` should be implemented from `default`

Comment (kurnevsky) [reply src/toxcore/state_format/old.rs]: There is no default value for `NoSpam`. `NoSpam::new` generates a random `NoSpam`. `Default` trait is supposed to give the same 'default' value every call, not random.

Comment (kpp) [reply src/toxcore/state_format/old.rs]: Did you get warnings from clippy about that?

Comment (kurnevsky) [reply src/toxcore/state_format/old.rs]: We have these warnings disabled:
```rust
#![cfg_attr(feature = "cargo-clippy", allow(clippy::new_without_default, clippy::new_without_default_derive))]
```
Actually I'm unsure regarding these default implementations. But I'd rather rename `new` to `random` and not provide `Default` trait implementation. Then we can reenable these warnings.

Comment (kpp) [reply src/toxcore/state_format/old.rs]: OK. In another PR.

**Pull request merged**